apps/smart-contracts/token/deploy/BlocklistTransferHook.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-console */
+import { DeployFunction } from 'hardhat-deploy/types'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+import { ChainId, getPrePOAddressForNetwork } from 'prepo-constants'
+import { utils } from 'prepo-hardhat'
+import { getNetworkByChainId } from 'prepo-utils'
+import dotenv from 'dotenv'
+
+dotenv.config({
+  path: '../.env',
+})
+
+const { assertIsTestnetChain } = utils
+
+const deployFunction: DeployFunction = async function deployBlocklistTransferHook({
+  ethers,
+  deployments,
+  getChainId,
+}: HardhatRuntimeEnvironment): Promise<void> {
+  const { deploy, getOrNull } = deployments
+  const deployer = (await ethers.getSigners())[0]
+  console.log(
+    'Running BlocklistTransferHook deployment script with',
+    deployer.address,
+    'as the deployer'
+  )
+  const currentChain = Number(await getChainId()) as ChainId
+  const currentNetwork = getNetworkByChainId(currentChain)
+  /**
+   * Make sure this script is not accidentally targeted towards a production environment.
+   * This can be temporarily removed if deploying to prod.
+   */
+  assertIsTestnetChain(currentChain)
+  const governanceAddress = getPrePOAddressForNetwork(
+    'GOVERNANCE',
+    currentNetwork.name,
+    process.env.GOVERNANCE
+  )
+  console.log('Governance for the current network is at:', governanceAddress)
+  const {
+    address: blocklistTransferHookAddress,
+    newlyDeployed: blocklistTransferHookNewlyDeployed,
+  } = await deploy('BlocklistTransferHook', {
+    from: deployer.address,
+    contract: 'BlocklistTransferHook',
+    deterministicDeployment: false,
+    args: [governanceAddress],
+    skipIfAlreadyDeployed: true,
+  })
+  if (blocklistTransferHookNewlyDeployed) {
+    console.log('Deployed BlocklistTransferHook to', blocklistTransferHookAddress)
+  } else {
+    console.log('Existing BlocklistTransferHook at', blocklistTransferHookAddress)
+  }
+  console.log('')
+}
+
+export default deployFunction
+
+deployFunction.dependencies = ['AccounList']
```

Comment:
```suggestion
deployFunction.dependencies = ['AccountList']
```

Typo aside, we should leave this out as a dependency. Dependencies make sense when there is a series of linear contract dependencies where each contract is only meant to be deployed once. 

We are going to need multiple different `AccountList` so dependencies don't make sense here.

Instead, we should add another section before deploying `BlocklistTransferHook`, to see if there is an `AccountList` deployed already specifically for the blocklist, similarly to how we do with `PPO`. The name we would fetch would be a name that we'd deploy an `AccountList` under, so maybe we'd do `getOrNull('Blocklist-AccountList')`.

This is why also, I would suggest writing the `AccountList` deploy script before hand, since a reviewer wouldn't even be able to run this without a `AccountList` deployment script.